buildSrc/src/main/java/Libs.kt

```diff
@@ -1,5 +1,20 @@
 import org.gradle.api.Project
 
+val Project.SharedLibs: SharedLibs
+    get() = SharedLibs(this)
+
+class SharedLibs(private val project: Project) {
```

Comment:
It's still slightly annoying that the user must know if the libs he wants to use is a Lib or a SharedLib, even though this concept is not really relevant at this level.

Reply:
The alternative is to stop using objects and use inner classes + properties for every dependencies even though they don't require project context.

I prefer this but I'm open if other people share your preference.